server/routes/changeEstablishment.ts

```diff
@@ -22,16 +22,23 @@ export default function routes(router: Router, supportedPrisonsService: Supporte
   get('/', establishmentSwitcherCheckMiddleware, async (req, res) => {
     const supportedPrisons = await supportedPrisonsService.getSupportedPrisons(res.locals.user?.username)
 
+    const referrer = (req.query?.referrer as string) ?? ''
```

Comment:
Check needs to happen anywhere you're pulling the referrer, stick it as a fn in utlis or something and just call it 🙂